File: 2478-longest-nice-subarray/longest-nice-subarray.cpp
class Solution {
public:
    int longestNiceSubarray(vector<int>& nums) {
        int sol=0;
        int l=0,mask=0;
        for(int r=0;r<nums.size();++r){
            while((mask & nums[r])!=0) mask^=nums[l++];
            mask|=nums[r];
            sol=max(sol,r-l+1);
        }
        return sol;
    }
};

File: 2478-longest-nice-subarray/longest-nice-subarray.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        n=len(nums)
        
        mask=0
        left=0
        sol=0
        
        for right in range(n):
            while mask&nums[right]!=0:
                mask^=nums[left]
                left+=1
            
            sol=max(sol,right-left+1)
            mask|=nums[right]
        
        return sol